File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        # die idee ist von links nach rechts zu laufne bis wir dei summe haben
        # danach von links das fenster zu verkleinern
        # wie berechnen wir hier dei summe des fenster -> supe einfach die berechenne einfach
        # mit der for die summe


        l = 0 
        sumsum = 0
        res = float("inf")
        for r in range(len(nums)):
            sumsum +=nums[r]

            while (target <= sumsum ):
                sumsum -= nums[l]
                res = min(res, r-l+1)
                l+=1
            
        return 0 if res == float("inf") else res

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums)
    {
        int len = nums.Length; 
        int[] res = new int [len];

        for(int i = 0; i < len; i++)
        {
            int prod = 1; 
            for(int j = 0; j< len; j++)
            {
             if(i != j) prod *=nums[j];
            }
            res[i] = prod;
        }

    return res; 
    }
}
